cmd/beacon-chain/flags/config.go

```diff
@@ -52,12 +54,20 @@ func ConfigureGlobalFlags(ctx *cli.Context) error {
 		cfg.SubscribeToAllSubnets = true
 	}
 
-	if ctx.Bool(SubscribeAllDataSubnets.Name) {
+	supernodeSet := ctx.Bool(SubscribeAllDataSubnets.Name)
+	semiSupernodeSet := ctx.Bool(SemiSupernode.Name)
+
+	// Ensure mutual exclusivity between supernode and semi-supernode modes
+	if supernodeSet && semiSupernodeSet {
+		return fmt.Errorf("cannot set both --%s and --%s flags; choose one mode", SubscribeAllDataSubnets.Name, SemiSupernode.Name)
```

Comment:
Should it be `Error` (the node continues to operate) or `Fatal` (the node stops operating).
IMO it should be `Fatal`.

Also, in the CLI help, IMO it should be noticed that both options cannot be used together.

Reply:
throwing an error will stop the node from starting all together

Reply:
https://github.com/OffchainLabs/prysm/pull/16029/commits/70bfe54012c8aa0cc6079011b5144b264bfb8ead update to description